ARCHITECTURE.md

```diff
@@ -0,0 +1,23 @@
+# Architecture Guide
+
+## Crate Organization
+The project is organized into two crates:
+
+1. `tpchgen`: The core library that implements the data generation logic for TPCH.
+2. `tpchgen-cli`: A CLI tool that uses the `tpchgen` library to generate TPCH data.
+
+## Dependencies
+
+The `tpchgen` crate is designed to be embeddable in as many locations as
+possible and thus has very minimal dependencies by design. For example, it does
+not depend on arrow or parquet crates or display libraries.
+
+The `tpchgen-cli` crate is designed to include many useful features, and thus
+has many more dependencies.
+
+## Speed
+
+Speed is a very important aspect of this project, and care has been taken to keep 
+the code as fast as possible, using some of the following techniques:
+1. Avoiding heap allocations during data generation
+2. Integer arithmetic instead of floating point arithmetic
```

Comment:
More of a technicality, but I think the speed up with the decimal stuff was from removing float formatting, and less around fp arithmetic.

I also think correctness trumps speed here, and the move to decimal was going for a more correct approach.

Reply:
I agree that the motivation for switching to decimal might not have been speed, but I thought it was a good example of things we have done to make this faster. How about this

```suggestion
2. Integer arithmetic and display instead of floating point arithmetic and display
```

Reply:
Works for me.